src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/ScannedComponentExtensions.cs

```diff
@@ -10,30 +10,30 @@
 namespace Microsoft.Sbom.Adapters.ComponentDetection;
 
 /// <summary>
-/// Extensions methods for <see cref="ScannedComponentWithLicense"/>.
+/// Extensions methods for <see cref="ExtendedScannedComponent"/>.
 /// </summary>
 public static class ScannedComponentExtensions
 {
     /// <summary>
-    /// Converts a <see cref="ScannedComponentWithLicense"/> to an <see cref="SbomPackage"/>.
+    /// Converts a <see cref="ExtendedScannedComponent"/> to an <see cref="SbomPackage"/>.
     /// </summary>
-    public static SbomPackage? ToSbomPackage(this ScannedComponentWithLicense component, AdapterReport report)
+    public static SbomPackage? ToSbomPackage(this ExtendedScannedComponent component, AdapterReport report)
     {
         return component.Component switch
         {
-            CargoComponent cargoComponent => cargoComponent.ToSbomPackage(component?.License),
+            CargoComponent cargoComponent => cargoComponent.ToSbomPackage(component?.LicenseConcluded),
```

Comment:
This is starting to get unwieldy. I would avoid passing parameters in and movethis logic into the extension methods.

You can check if `cargoComponent` is an `ExtendedScannedComponent` for example:

```cs
if (mavenComponent is ExtendedScannedComponent extendedScannedComponent)
{
    // ...
}
```

In the extension methods